contracts/governance/StakingRewards/StakingRewardsOs.sol

```diff
@@ -0,0 +1,256 @@
+     * @param _osSOV osSOV token address
+     * @param _staking StakingProxy address should be passed
+     * */
+    function initialize(
+        address _osSOV,
+        IStaking _staking,
+        uint256 _averageBlockTime
+    ) external onlyOwner initializer {
+        require(_osSOV != address(0), "Invalid OsSOV Address");
+        require(Address.isContract(_osSOV), "OsSOV is not a contract");
+        osSOV = IERC20Mintable(_osSOV);
+        staking = _staking;
+        rewardsProgramStartTime = staking.timestampToLockDate(block.timestamp);
+        maxDuration = 15 * TWO_WEEKS;
+        deploymentBlock = _getCurrentBlockNumber();
+        averageBlockTime = _averageBlockTime;
+    }
+
+    /**
+     * @notice Stops the current rewards program.
+     * @dev Users will only get rewards up to the stop block
+     * */
+    function stop() external onlyOwner {
+        require(stopBlock == 0, "Already stopped");
+        stopBlock = _getCurrentBlockNumber();
+        stopRewardsTimestamp = staking.timestampToLockDate(block.timestamp);
+    }
+
+    /**
+     * @notice Collect rewards
+     * @dev User calls this function to collect osSOV staking rewards accrued by this contract
+     * The weighted stake is calculated using getPriorWeightedStake. Block number sent to the functon
+     * must be a finalised block, hence we deduct 1 from the current block. User is only allowed to withdraw
+     * after intervals of 14 days.
+     * @param _startTime The time from which to start the staking rewards calculation
+     * The issue is that we can only run for a max duration and if someone stakes for the
+     * first time after the max duration is over, the reward will always return 0. Thus, we need to restart
+     * from the duration that elapsed without generating rewards.
+     * */
+    function claimReward(uint256 _startTime) external {
+        require(
+            stopBlock == 0 || stakerNextWithdrawTimestamp[msg.sender] < stopRewardsTimestamp,
+            "Entire reward already paid"
+        );
+        (uint256 withdrawTimestamp, uint256 amount) = _getStakerCurrentReward(
+            msg.sender,
+            true,
+            _startTime
+        );
+        require(withdrawTimestamp > 0 && amount > 0, "No valid reward");
+        stakerNextWithdrawTimestamp[msg.sender] = withdrawTimestamp;
+        _payReward(msg.sender, amount);
+    }
+
+    /**
+     * @notice Changes average block time - based on blockchain
+     * @dev If average block time significantly changes, we can update it here and use for block number calculation
+     */
+    function setAverageBlockTime(uint256 _averageBlockTime) external onlyOwner {
+        averageBlockTime = _averageBlockTime;
+    }
+
+    /**
+     * @notice This function computes the last staking checkpoint and calculates the corresponding
+     * block number using the average block time which is then added to the mapping `checkpointBlockNumber`.
+     */
+    function setBlock() external {
```

Comment:
Since this function can be called by anyone, I am assuming this function needs to be called periodically?